app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/WordCount.java

```diff
@@ -48,10 +48,14 @@ public static int getAverageWordCountReceived(ArrayList<Sms> smsList) {
 
     private static int calculateAverage(ArrayList<Integer> numbers) {
         int sum = 0;
-        for (int i = 0; i < numbers.size(); i++) {
+        int arraySize = numbers.size();
+        for (int i = 0; i < arraySize; i++) {
             sum = sum + numbers.get(i);
         }
-        int average = sum / numbers.size();
-        return average;
+        if(sum != 0){
```

Comment:
really you should be checking arraySize, but this would be fine (as in this fixes the bug too)